ergo/distributions/histogram.py

```diff
@@ -104,8 +104,54 @@ def sample(self):
     def rv(self):
         raise NotImplementedError
 
-    def normalize(self):
-        return HistogramDist(self.logps, 0, 1)
+    def normalize(self, true_scale: scale.Scale = None):
+        """
+        Normalize the histogram onto [0,1]
+
+        Setting a true_scale allows you to express that the histogram to be normalized
+        did not cover the entire scale of interest.
+
+        E.g. -- imagine my histogram only has probability mass from 2 to 5,
```

Comment:
1. I implemented this in `normalize` because that's the function I needed to call, and it's a fairly sensible place to put it. But arguably, this functionality should live on its own as `zero_pad` or something like that
2. The strategy here is to pad out the histogram with bins of zero probability, increasing the number of bins. Some possible alternatives:
a. keep the same number of bins, squeeze the probability into the middle, pad with zeros. I didn't want to do this because we'd have to destroy information (toss out some of the values we already have)
b. refactor HistogramDist so that it no longer assumes that the bins are evenly spaced. Then we'd only have to add a couple bins on either side of the dist to zero pad it
c. refactor HistogramDist and probably some other things (`CrossEntropyCondition`?) so that the HistogramDist expresses the fact that the histogram doesn't take up the entire true_scale in some other way than via its bins

@stuhlmueller, any thoughts on 1 or 2?

Reply:
2b. I think Dan is working on this as part of log scale PR.

https://github.com/oughtinc/ergo/compare/log_questions